app/page.tsx

```diff
@@ -2,29 +2,201 @@
+  const [quizzes, setQuizzes] = useState<LiveClasses[]>([]);
+  const baseUrl = process.env.NEXT_PUBLIC_AF_QUIZ_URL;
+  const apiKey = process.env.NEXT_PUBLIC_AF_QUIZ_API_KEY;
+  const userID = process.env.NEXT_PUBLIC_AF_QUIZ_USER_ID;
+
+  const fetchSessionOccurrencesAndDetails = async () => {
+    try {
+      const sessionOccurrenceData = await getSessionOccurrences();
+      const currentDate = new Date();
+      currentDate.setHours(0, 0, 0, 0);
+      const endOfWeek = new Date();
+      endOfWeek.setDate(endOfWeek.getDate() + (6 - endOfWeek.getDay()));
+
+      const todaySessions = [];
+
+      for (const sessionOccurrence of sessionOccurrenceData) {
+        const sessionStartTime = new Date(sessionOccurrence.start_time);
+
+        if (sessionStartTime > currentDate && sessionStartTime <= endOfWeek) {
+          try {
+            const sessionDetail = await getSessions(sessionOccurrence.session_fk);
+            if (isSameDay(sessionStartTime, currentDate)) {
+              todaySessions.push({
+                sessionOccurrence,
+                sessionDetail,
+              });
+            }
+          } catch (error) {
+            console.error("Error fetching session details for session ID:", sessionOccurrence.session_fk, error);
+          }
+        }
+      }
+      const liveClassesToday = todaySessions.filter(data => data.sessionDetail.platform === "meet");
+      const quizzesToday = todaySessions.filter(data => data.sessionDetail.platform === "quiz");
+
+      setLiveClasses(liveClassesToday);
+      setQuizzes(quizzesToday);
+    } catch (error) {
+      console.error("Error in fetching Live Classes:", error);
+    }
+  }
+
+
+  function isSameDay(date1: Date, date2: Date): boolean {
+    return (
+      date1.getDate() === date2.getDate() &&
+      date1.getMonth() === date2.getMonth() &&
+      date1.getFullYear() === date2.getFullYear()
+    );
+  }
+
+
+  function formatTime(dateTimeStr: string) {
+    const date = new Date(dateTimeStr);
+    const hours = String(date.getUTCHours()).padStart(2, "0");
+    const minutes = String(date.getUTCMinutes()).padStart(2, "0");
+    return `${hours}:${minutes}`;
+  }
+
+  function renderButton(data: { sessionOccurrence: any, sessionDetail: any }) {
+    const currentTime = new Date();
+    const sessionTimeStr = formatTime(data.sessionOccurrence.start_time);
+    const currentTimeStr = formatTime(currentTime.toISOString());
+
+    if (data.sessionDetail.platform === 'meet') {
+      if (sessionTimeStr <= currentTimeStr) {
+        return (
+          <Link href={`https://${data.sessionDetail.platform_link}`} target="_blank">
+            <PrimaryButton
+              className="bg-primary text-white text-sm rounded-lg w-14 h-8 mr-4 shadow-md shadow-slate-400">JOIN</PrimaryButton>
+          </Link>
+        );
+      } else {
+        return (
+          <p className="text-sm italic font-normal mr-6">
+            Starts on <br />
```

Comment:
Starts at

Reply:
![image](https://github.com/avantifellows/gurukul/assets/107760942/1d26060f-9a1e-4cca-9ed9-0f784b5ab366)


In the mockups, it says starts on XD. Anyways, I'll change it